exprs.go

```diff
@@ -0,0 +1,968 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package iceberg
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+//go:generate stringer -type=Operation -linecomment
+
+// Operation is an enum used for constants to define what operation a given
+// expression or predicate is going to execute.
+type Operation int
+
+const (
+	// do not change the order of these enum constants.
+	// they are grouped for quick validation of operation type by
+	// using <= and >= of the first/last operation in a group
+
+	OpTrue  Operation = iota // True
+	OpFalse                  // False
+	// unary ops
+	OpIsNull  // IsNull
+	OpNotNull // NotNull
+	OpIsNan   // IsNaN
+	OpNotNan  // NotNaN
+	// literal ops
+	OpLT            // LessThan
+	OpLTEQ          // LessThanEqual
+	OpGT            // GreaterThan
+	OpGTEQ          // GreaterThanEqual
+	OpEQ            // Equal
+	OpNEQ           // NotEqual
```

Comment:
I'm now seeing `predicates.go`, if this is just internal, feel free to ignore the comment above.

Reply:
So the comment lines on those are for use with `stringer` to auto-generate the `String()` method for the enum values. By adding the `-linecomment` option it uses the comment as the string that gets returned instead of just creating a string from the value. e.g. `OpLessThan.String()` would return `"OpLessThan"`, but by using `-linecomment` and adding `// LessThan` to it, it returns `"LessThan"`.

Does that make sense?

Reply:
Yes, that makes sense. Thanks for the explanation 👍